docs: update README.md

MichaelCurrin opened the pull request:
- [x] This PR is for the `dev` branch rather than for the `release` branch.
- [x] This PR is compliant with the [other contributing guidelines](https://github.com/nodemcu/nodemcu-firmware/blob/dev/CONTRIBUTING.md) as well (if not, please describe why).

deining committed:
README 'spissimg' tool: fix broken link (#3430)

Fixing several typos

MichaelCurrin committed:
docs: update README.md

MichaelCurrin committed:
docs: update shield

MichaelCurrin commented:
I noticed the Travis link goes here

https://travis-ci.org/github/nodemcu/nodemcu-firmware

Which is deprecated

and recommends going here. So I used this

https://travis-ci.com/github/nodemcu/nodemcu-firmware

marcelstoer commented:
Some of those changes are subjective. As for Travis, we should actually remove `.travis.yml` (as we don't need it anymore) and instead add the shield for our GitHub workflow.

And this really should land on `dev` instead of `release` as the template requires.

MichaelCurrin committed:
docs: replace badge

MichaelCurrin commented on README.md:
```diff
@@ -1,31 +1,31 @@
+## Programming Model
 
-The NodeMCU programming model is similar to that of [Node.js](https://en.wikipedia.org/wiki/Node.js), only in Lua. It is asynchronous and event-driven. Many functions, therefore, have parameters for callback functions. To give you an idea what a NodeMCU program looks like study the short snippets below. For more extensive examples have a look at the [`/lua_examples`](lua_examples) folder in the r ...
+The NodeMCU programming model is similar to that of [Node.js](https://en.wikipedia.org/wiki/Node.js), only in Lua. It is asynchronous and event-driven. Many functions, therefore, have parameters for callback functions. To give you an idea what a NodeMCU program looks like study the short snippets below. For more extensive examples have a look at the [lua_examples](/lua_examples/) directory.
```
```suggestion
The NodeMCU programming model is similar to that of [Node.js](https://en.wikipedia.org/wiki/Node.js), only in Lua. It is asynchronous and event-driven. Many functions, therefore, have parameters for callback functions. To give you an idea what a NodeMCU program looks like study the short snippets below. For more extensive examples have a look at the [`/lua_examples`](/lua_examples/) folder in the repository on GitHub.
```

MichaelCurrin commented on README.md:
```diff
@@ -1,31 +1,31 @@
+- More than **70 built-in C modules** and **close to 20 Lua modules**
 - Firmware available with or without floating point support (integer-only uses less memory)
-- Up-to-date documentation at [https://nodemcu.readthedocs.io](https://nodemcu.readthedocs.io)
+- Up-to-date documentation at [NodeMCU documentation]
```
```suggestion
- Up-to-date documentation at [https://nodemcu.readthedocs.io](https://nodemcu.readthedocs.io)
```

MichaelCurrin commented on README.md:
```diff
@@ -44,27 +44,36 @@ wifi.setmode(wifi.STATION)
-# Documentation
+## Documentation
+
+The entire [NodeMCU documentation] is maintained right in this repository in [docs](/docs/). The fact that the API documentation is maintained in the same repository as the code that _provides_ the API ensures consistency between the two. 
```
```suggestion
The entire [NodeMCU documentation](https://nodemcu.readthedocs.io) is maintained right in this repository at [/docs](/docs/). The fact that the API documentation is maintained in the same repository as the code that _provides_ the API ensures consistency between the two.

With every commit the documentation is rebuilt by Read the Docs and thus transformed from terse Markdown into a nicely browsable HTML site at [https://nodemcu.readthedocs.io](https://nodemcu.readthedocs.io).
```

MichaelCurrin commented on README.md:
```diff
@@ -44,27 +44,36 @@ wifi.setmode(wifi.STATION)
+
+The entire [NodeMCU documentation] is maintained right in this repository in [docs](/docs/). The fact that the API documentation is maintained in the same repository as the code that _provides_ the API ensures consistency between the two. 
+
+With every commit, the documentation is rebuilt by _Read the Docs_ and thus transformed from terse Markdown into a convenient HTML site hosted here - [NodeMCU documentation].
```
```suggestion
```

MichaelCurrin commented on README.md:
```diff
@@ -44,27 +44,36 @@ wifi.setmode(wifi.STATION)
+## Support
 
-# Support
+See the [Support](https://nodemcu.readthedocs.io/en/release/support/) doc.
```
```suggestion
See [https://nodemcu.readthedocs.io/en/release/support/](https://nodemcu.readthedocs.io/en/release/support/).
```

MichaelCurrin committed:
docs: revert README.md changes

MichaelCurrin commented on README.md:
```diff
@@ -1,31 +1,31 @@
+## Programming Model
 
-The NodeMCU programming model is similar to that of [Node.js](https://en.wikipedia.org/wiki/Node.js), only in Lua. It is asynchronous and event-driven. Many functions, therefore, have parameters for callback functions. To give you an idea what a NodeMCU program looks like study the short snippets below. For more extensive examples have a look at the [`/lua_examples`](lua_examples) folder in the r ...
+The NodeMCU programming model is similar to that of [Node.js](https://en.wikipedia.org/wiki/Node.js), only in Lua. It is asynchronous and event-driven. Many functions, therefore, have parameters for callback functions. To give you an idea what a NodeMCU program looks like study the short snippets below. For more extensive examples have a look at the [`/lua_examples`](/lua_examples/) folder in the ...
```
```suggestion
The NodeMCU programming model is similar to that of [Node.js](https://en.wikipedia.org/wiki/Node.js), only in Lua. It is asynchronous and event-driven. Many functions, therefore, have parameters for callback functions. To give you an idea what a NodeMCU program looks like study the short snippets below. For more extensive examples have a look at the [`/lua_examples`](lua_examples) folder in the repository on GitHub.
```

MichaelCurrin replied:
```suggestion
The NodeMCU programming model is similar to that of [Node.js](https://en.wikipedia.org/wiki/Node.js), only in Lua. It is asynchronous and event-driven. Many functions, therefore, have parameters for callback functions. To give you an idea what a NodeMCU program looks like study the short snippets below. For more extensive examples have a look at the [`/lua_examples`](lua_examples) folder in the repository on GitHub.
```

MichaelCurrin commented on README.md:
```diff
@@ -44,27 +44,37 @@ wifi.setmode(wifi.STATION)
-# Documentation
+## Documentation
+
+The entire [NodeMCU documentation](https://nodemcu.readthedocs.io) is maintained right in this repository at [/docs](/docs/). The fact that the API documentation is maintained in the same repository as the code that _provides_ the API ensures consistency between the two.
```
```suggestion
The entire [NodeMCU documentation](https://nodemcu.readthedocs.io) is maintained right in this repository at [/docs](docs). The fact that the API documentation is maintained in the same repository as the code that _provides_ the API ensures consistency between the two.
```

MichaelCurrin commented on README.md:
```diff
@@ -44,27 +44,37 @@ wifi.setmode(wifi.STATION)
+
+The entire [NodeMCU documentation](https://nodemcu.readthedocs.io) is maintained right in this repository at [/docs](/docs/). The fact that the API documentation is maintained in the same repository as the code that _provides_ the API ensures consistency between the two.
+
+With every commit the documentation is rebuilt by Read the Docs and thus transformed from terse Markdown into a nicely browsable HTML site at [https://nodemcu.readthedocs.io](https://nodemcu.readthedocs.io).
```
```suggestion
```

MichaelCurrin commented on README.md:
```diff
@@ -44,27 +44,37 @@ wifi.setmode(wifi.STATION)
+## Documentation
+
+The entire [NodeMCU documentation](https://nodemcu.readthedocs.io) is maintained right in this repository at [/docs](/docs/). The fact that the API documentation is maintained in the same repository as the code that _provides_ the API ensures consistency between the two.
+
```
```suggestion
```

MichaelCurrin committed:
docs: revert README.md changes

MichaelCurrin committed:
docs: revert README.md changes

MichaelCurrin commented on README.md:
```diff
@@ -44,27 +44,35 @@ wifi.setmode(wifi.STATION)
-# License
+## License
+
+Licensed under [MIT](/LICENSE) © [zeroday](https://github.com/NodeMCU)/[nodemcu.com](http://nodemcu.com/index_en.html)
```
```suggestion
[MIT](/LICENSE) © [zeroday](https://github.com/NodeMCU)/[nodemcu.com](http://nodemcu.com/index_en.html)
```

MichaelCurrin commented on README.md:
```diff
@@ -44,27 +44,35 @@ wifi.setmode(wifi.STATION)
+Licensed under [MIT](/LICENSE) © [zeroday](https://github.com/NodeMCU)/[nodemcu.com](http://nodemcu.com/index_en.html)
 
-[MIT](https://github.com/nodemcu/nodemcu-firmware/blob/release/LICENSE) © [zeroday](https://github.com/NodeMCU)/[nodemcu.com](http://nodemcu.com/index_en.html)
+[NodeMCU documentation]: https://nodemcu.readthedocs.io
```
```suggestion
```

MichaelCurrin committed:
docs: revert README.md changes

MichaelCurrin commented:
Thanks. I've removed the more subject changes.

Now it's mostly about making H1s into H2s (accessibility scores such in SEO engines will penalize you will having more than one H1 on a page).

I've replaced with a GH Actions badge.

Sorry I misunderstood how the dev/release bit worked.

MichaelCurrin commented on README.md:
```diff
@@ -44,27 +44,33 @@ wifi.setmode(wifi.STATION)
-A new tag is created every time `dev` is merged back to `release`. They are listed in the [releases section here on GitHub](https://github.com/nodemcu/nodemcu-firmware/releases). Tag names follow the \<SDK-version\>-release_yyyymmdd pattern.
+A new tag is created every time the `dev` branch is merged back to `release`. They are listed in this repo's [releases](https://github.com/nodemcu/nodemcu-firmware/releases).
+
+Tag names follow the `<SDK-version>-release_yyyymmdd` pattern.
```
This is on its own line otherwise when viewing the page on GitHub the codeblock wraps across to a new line in the middle of the line so it looks weird.

MichaelCurrin commented on README.md:
```diff
@@ -1,29 +1,29 @@
-# NodeMCU 3.0.0
+# NodeMCU Firmware 3.0.0
```
```suggestion
# NodeMCU 3.0.0
```

MichaelCurrin committed:
docs: revert README.md changes

MichaelCurrin commented on README.md:
```diff
@@ -1,29 +1,29 @@
 # NodeMCU 3.0.0
+> Lua-based firmware for ESP8266 WiFi SOC
```
Having a description directly after the heading followed by badges is a common pattern on repos.

HHHartmann commented:
do we want to also take this PR?
Why is ist against release and not dev?

marcelstoer commented:
I'll clean this up and rebase on `dev`.

marcelstoer committed:
Merge branch 'dev' into patch-1
